js/packages/teams-dialogs/src/prompts/teamsBotSsoPrompt.ts

```diff
@@ -0,0 +1,337 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Dialog } from "../dialog";
+import { TurnContext, TeamsChannelAccount, TeamsInfo, CardFactory, TokenExchangeResource, OAuthCard, ActionTypes, MessageFactory, Activity, ActivityTypes, verifyStateOperationName, tokenExchangeOperationName, TokenExchangeInvokeRequest, StatusCodes } from '@microsoft/teams-core';
+import { v4 as uuidv4 } from 'uuid';
+import { PromptRecognizerResult } from "./prompt";
+import { DialogContext } from "../dialogContext";
+import jwt_decode from "jwt-decode";
+import { ConfidentialClientApplication, NodeAuthOptions } from "@azure/msal-node";
+
+const invokeResponseType = "invokeResponse";
+
+type TeamsBotSsoPromptAppCredentials = {
+  /**
+   * Hostname of AAD authority.
+   */
+  authorityHost: string;
+
+  /**
+   * The client (application) ID of an App Registration in the tenant
+   */
+  clientId: string;
+
+  /**
+   * AAD tenant id
+   *
+   * @readonly
+   */
+  tenantId: string;
+} & (
+    | {
+      /**
+       * Secret string that the application uses when requesting a token. Only used in confidential client applications. Can be created in the Azure app registration portal.
+       */
+      clientSecret: string;
+      certificateContent?: never;
+    }
+    | {
+      clientSecret?: never;
+      /**
+       * The content of a PEM-encoded public/private key certificate.
+       *
+       * @readonly
+       */
+      certificateContent: string;
+    }
+  );
+
+export interface TeamsBotSsoPromptSettings {
+  scopes: string[];
+  timeout?: number;
+  endOnInvalidMessage?: boolean;
+  appCredentials: TeamsBotSsoPromptAppCredentials;
```

Comment:
Move this out of the prompt setting and put to prompt constructor: we should ensure minimum changes to existing code